### PR TITLE
Fix/13818 & 13819 mask state UI and screenshot tests

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
Mask State UI and Screenshot tests

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13818
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13819

## Screenshots
Screenshot 2022-09-23 at 14 13 46

Screenshot 2022-09-23 at 14 10 33

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift [hunk 1]
```diff
@@ -3,6 +3,7 @@
 //
 
 import HealthCertificateToolkit
+import Foundation
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift:6]: Do we need to add this?

File: src/xcode/ENA/ENAUITests/HealthCertificate/ENAUITests_13_CreateHealthCertificate.swift [hunk 2]
```diff
@@ -640,6 +649,7 @@ class ENAUITests_13_CreateHealthCertificate: CWATestCase {
 	
 	func test_screenshot_maskOptionalForFederalStateBW() throws {
 		app.setLaunchArgument(LaunchArguments.healthCertificate.testCertificateRegistered, to: true)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/HealthCertificate/ENAUITests_13_CreateHealthCertificate.swift:651]: Do we need still need testCertificateRegistered? or we can just use firstAndSecondHealthCertificate and isMaskOptionalStateEnabled ?

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Overall looks good to me, since the tests are running, can we check all the elements in the screenshot tests? or perhaps can we add another UI tests which can validates if the values on detail screen are correct.

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift:6]: Yes!

File: src/xcode/ENA/ENAUITests/HealthCertificate/ENAUITests_13_CreateHealthCertificate.swift (hunk 2 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENAUITests/HealthCertificate/ENAUITests_13_CreateHealthCertificate.swift]: Good Catch! changed it to firstAndSecondHealthCertificate.

**Comment by puneetmahali1** [comment]
> Overall looks good to me, since the tests are running, can we check all the elements in the screenshot tests? or perhaps can we add another UI tests which can validates if the values on detail screen are correct.

Sure, I have checked now all UI elements and already add the one more UI test now we have both MaskState (Optional & Required) UI tests.

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift [hunk 3]
```diff
@@ -314,6 +315,106 @@ extension HealthCertificateService {
...
+			vaccinationState: dccWalletInfo.vaccinationState,
+			maskState: DCCMaskState(visible: true, badgeText: badgeText, titleText: titleText, subtitleText: subtitleText, longText: longText, faqAnchor: "maskstate", identifier: .maskOptional),
+			boosterNotification: dccWalletInfo.boosterNotification,
+			mostRelevantCertificate: dccWalletInfo.mostRelevantCertificate,
+			verification: dccWalletInfo.verification,
+			validUntil: dccWalletInfo.validUntil,
+			certificateReissuance: dccWalletInfo.certificateReissuance,
+			certificatesRevokedByInvalidationRules: dccWalletInfo.certificatesRevokedByInvalidationRules
+		)
+	}
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift:417]:
```suggestion
	}

```

**Review by naveeddotio — APPROVED** [review]

**Comment by naveeddotio** [comment]
> > Overall looks good to me, since the tests are running, can we check all the elements in the screenshot tests? or perhaps can we add another UI tests which can validates if the values on detail screen are correct.
> 
> Sure, I have checked now all UI elements and already add the one more UI test now we have both MaskState (Optional & Required) UI tests.

I was referring to a UI test for detail screen

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift (hunk 3 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService+UITestingMethods.swift:417]: Thanks, done!

**Review by 30mar — APPROVED** [review]

**Pull request merged**